plugin/src/main/java/jenkins/plugins/openstack/compute/JCloudsSlave.java

```diff
@@ -159,10 +159,15 @@ protected Object readResolve() {
                     .jvmOptions(Util.fixEmpty(jvmOptions))
             ;
 
-            LauncherFactory lf = "SSH".equals(slaveType)
-                    ? new LauncherFactory.SSH(credentialsId)
-                    : LauncherFactory.JNLP.JNLP
-            ;
+            LauncherFactory lf = null;
+            if ("SSH".equals(slaveType)) {
+                lf = new LauncherFactory.SSH(credentialsId);
+            } else if ("JNLP".equals(slaveType)) {
+                lf = LauncherFactory.JNLP.JNLP;
+            } else if ("KAFKA".equals(slaveType)) {
```

Comment:
There is no need to add these into `readResolve` methods. They are only used to migrate old config to new one. And since kafka is just being added, it does not exist in user configs.